### DISCUSSION TITLE
Unable to import StatusCode from parcels

### DISCUSSION THREAD

**Discussion by jwongala** [Parcels-code/Parcels#1708] [Q&A] [answered]
Hello! 

I am trying to use to the StatusCode `CheckOutOfBounds` but when I got to import it from the version of parcels I have downloaded I get the error seen in the screenshot below. Any help is greatly appreciated! 
Thank you!

Jennifer 

![Screenshot 2024-09-16 at 5 39 14 PM](https://github.com/user-attachments/assets/766c2133-d5c0-4e43-9676-423e5fe6d3c3)

**Comment by erikvansebille** [comment]
@VeckoTheGecko, can you have a lok what's going on here?

**Comment by VeckoTheGecko** [comment, marked as answer]
Hi @jwongala . `from parcels import StatusCode` works for me. What version of Parcels are you running?

```py
import parcels
print(parcels.__version__) # let us know what this says
```

It also looks like this code you're running is from an older version of parcels, as we no longer have `ErrorCode` or `plotTrajectoriesFile`. Does the code you're working from mention the version of Parcels they were using?

**Comment by jwongala** [comment]
Hi @VeckoTheGecko Thanks! After running the code you sent, it says I am using version 2.4.1. The code I am working on doesn't mention the version, but I originally started using it in 2020. I thought I updated Parcels, but I don't think I did it correctly. What directions should I follow to update?

**Reply by VeckoTheGecko** [reply]
If you run `conda update parcels` and then try run the version code again, you should see a version >3 (probably 3.0.5). Updating you code is another matter though. The major changes should be between v2 and v3, which you can see in our [v3 release notes](https://github.com/OceanParcels/Parcels/releases/tag/v3.0.0).

**Reply by jwongala** [reply]
Hi @VeckoTheGecko it still says I am running on version 2.4.1 after updating (first screenshot). I attached a screenshot of my terminal when I just updated too. I am not sure what is wrong. Thank you for all of your help. 

<img width="523" alt="Screenshot 2024-09-18 at 1 37 34 PM" src="https://github.com/user-attachments/assets/37d63749-290c-46f1-8ab3-40fbdbb1fa3a">

![Screenshot 2024-09-18 at 1 39 02 PM](https://github.com/user-attachments/assets/13cf7ead-475b-40ca-9533-a46980a2bb57)

**Reply by VeckoTheGecko** [reply]
> it still says I am running on version 2.4.1 after updating (first screenshot)

In that case I think you're probably best off just creating a new environment and installing a new copy of Parcels (likely its due to a conflict with your other packages). `my_env` can be any name that you want

```console
conda create -n my_env
conda activate my_env
conda install -c conda-forge parcels
```

**Comment by jwongala** [comment]
Hi @VeckoTheGecko I just created a new environment called `parcels_model` and installed a new copy of parcels, but it still doesn't seem to be working. I've attached screenshots. 

![Screenshot 2024-09-18 at 2 15 23 PM](https://github.com/user-attachments/assets/fb646dc7-1bfc-4fd3-aeaa-03499c63bdb2)

![Screenshot 2024-09-18 at 2 16 04 PM](https://github.com/user-attachments/assets/f04fd636-c1c5-47b6-8e7d-cdce27b9f343)

**Reply by VeckoTheGecko** [reply]
There much be a problem where Jupyter can't see the environment that you're working within (and also from "All requested packages already installed" indicates that the environment you created already came with parcels and wasn't blank - where the commands I suggested were for a blank environment).

Unfortunately the management of conda environments and getting them to link with Jupyter Notebook isn't something that we're able to provide support on. I recommend reaching out to colleagues that have used Anaconda before, going through resources such as [the conda user guide](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html), discussing with your institutions IT support, or reaching out to other online communities that are better suited for conda/general Python help (e.g., the help channels in [Python Discord](https://discord.com/invite/python))

**Reply by jwongala** [reply]
Hi @VeckoTheGecko, Thanks for the info. I will contact someone here at my university to see if I can get some help. 

Are there other versions or methods I could use similar to `CheckOutOfBounds` developed with the 2.4.1 parcels version that I could use? @erikvansebille 

Thank you all for your help!

**Reply by VeckoTheGecko** [reply]
Dealing with out of bound particles in Parcels v2.4.1 is detailed in the following tutorial https://docs.oceanparcels.org/en/v2.4.2/examples/tutorial_Agulhasparticles.html#Agulhas-region-particles
